### DISCUSSION TITLE
blog/zh/how-to-create-your-blog-for-free-by-hugo-ladder-in-30min/

### DISCUSSION THREAD

**Discussion by giscus** [guangzhengli/blog-comments#8] [Announcements]
# blog/zh/how-to-create-your-blog-for-free-by-hugo-ladder-in-30min/

Ladder@本文是关于如何搭建免费的独立博客系列的第二篇，包括如何使用本博客主题 hugo-ladder，如何集成评论和统计功能等。如果后续打算使用该主题来搭建和运营博客，建议跟随本篇文章来搭建。

https://guangzhengli.com/blog/zh/how-to-create-your-blog-for-free-by-hugo-ladder-in-30min/

**Comment by zhuzhuyan93** [comment]
请问你的博客的中文字体是啥

**Reply by guangzhengli** [reply]
[霞鹜文楷](https://github.com/lxgw/LxgwWenKai)

**Reply by zhuzhuyan93** [reply]
还有一个问题，giscus的配置好像没有适配夜间模式，有考虑适配一下嘛

**Reply by guangzhengli** [reply]
现在把Giscus切换夜间模式的事件放在页面渲染的时候，所以需要切换夜间模式后刷新下页面😂。确实可以考虑后续优化一下

**Comment by AlainHsu** [comment]
写的很棒୧(๑•̀◡•́๑)૭
一直都是用 Github Page 部署，没弄过域名。看了你的文章非常推荐 Cloudflare，试了下链接到 Cloudflare Page，确实很方便，还顺手买了域名哈哈。

**Comment by Buu98** [comment]
运行报错，是不是对版本有要求？

PS D:\hugo\hugo-demo\hugo-ladder-exampleSite> hugo server -D
Start building sites …
hugo v0.111.3-5d4eb5154e1fed125ca8e9b5a0315c4180dab192+extended windows/amd64 BuildDate=2023-03-12T11:40:50Z VendorInfo=gohugoio
Error: Error building site: "D:\hugo\hugo-demo\hugo-ladder-exampleSite\content\blog\zh\rich-content.zh.md:14:1": failed to extract shortcode: template for shortcode "toggle" not found
Built in 11 ms

**Reply by guangzhengli** [reply]
应该不是版本的原因，目前主题中是有 toggle 这个 shortcode 的，用来实现点击展开隐藏的功能。但我在本地无法复现这个问题，你可以暂时搜索 `\content\blog\` 博客文件夹下删除所有的 `{{< /toggle>}}` 关键字暂时禁用  toggle 这个 shortcode，影响不大。

**Reply by ZimaBlueee** [reply]
最新版的hugo遇到了同样的问题。。。

另外，拉了博主的项目，启动报了另一个错：
```
ERROR Failed to get JSON resource "https://publish.twitter.com/oembed?dnt=false&url=https%3A%2F%2Ftwitter.com%2Figuangzhengli%2Fstatus%2F1668059021390266369": Get "https://publish.twitter.com/oembed?dnt=false&url=https%3A%2F%2Ftwitter.com%2Figuangzhengli%2Fstatus%2F1668059021390266369": net/http: TLS handshake timeout
```

最后我把博主的博客全删完了，只留下我自己新建的一篇博客，访问本地1313端口结果显示404，折腾了一晚上实在不知道问题出在哪里。。。

**Reply by ZimaBlueee** [reply]
404的启动日志：
```
(base) zima@cj:~/code/webstorm/blog$ hugo server -D
Watching for changes in /home/zima/code/webstorm/blog/{archetypes,assets,content,data,i18n,static}
Watching for config changes in /home/zima/code/webstorm/blog/hugo.yml
Start building sites … 
hugo v0.120.4-f11bca5fec2ebb3a02727fb2a5cfb08da96fd9df+extended linux/amd64 BuildDate=2023-11-08T11:18:07Z VendorInfo=gohugoio

WARN  found no layout file for "html" for layout "archives" for kind "page": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for layout "art" for kind "page": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for kind "page": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for layout "tags" for kind "page": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for kind "home": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for layout "sponsors" for kind "section": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for kind "taxonomy": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for kind "term": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for layout "guestbook" for kind "page": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for layout "blog" for kind "page": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
WARN  found no layout file for "html" for kind "section": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.

                   | EN  
-------------------+-----
  Pages            | 16  
  Paginator pages  |  0  
  Non-page files   |  0  
  Static files     |  3  
  Processed images |  0  
  Aliases          |  0  
  Sitemaps         |  1  
  Cleaned          |  0  
```

**Reply by wangqiqi66** [reply]
切换了版本就好了

**Comment by helloChenLei** [comment]
我想讲 hexo 迁移到 hugo ，使用你的主题；于是我使用 https://github.com/guangzhengli/hugo-ladder-exampleSite 直接部署，全程按照要求来的，但是出现了下列[错误](https://github.com/helloChenLei/helloChenLei.github.io/actions/runs/5745582219/job/15573790912) ，排查可能是权限问题，也可能是gh-pages分支没建立，最后都没成功😂，想请教下，如何解决呀
Push the commit or tag
  /usr/bin/git push origin gh-pages
  remote: Permission to helloChenLei/helloChenLei.github.io.git denied to github-actions[bot].
  fatal: unable to access 'https://github.com/helloChenLei/helloChenLei.github.io.git/': The requested URL returned error: 403
  Error: Action failed with "The process '/usr/bin/git' failed with exit code 128"

**Reply by helloChenLei** [reply]
自问自答一下，目前博客已经全部迁移过来，上面的问题也解决了；同时总结了一篇文章：[博客从 Hexo 迁移到 Hugo 的过程](https://www.demochen.com/posts/20230804/ ) ，针对主题做了很多定制化修改，比如新增目录是否显示、调整文章URL、分类等等，感谢博主的贡献～

**Reply by hui61** [reply]
发现了一个问题，文章字数和阅读预计时间应该是不对的，目前我的[Blog](https://hui61.github.io)还没有发现解决办法，我只能去掉文章字数和阅读预计时间。

**Comment by hui61** [comment]
这是啥问题呀
Error: error building site: assemble: "/Users/xxx/Projects/mine/hugo-blog/my_website/content/blog/zh/how-to-create-your-blog-for-free-by-hugo-ladder-in-30min.zh.md:1:1": got closing shortcode, but none is open

**Reply by hui61** [reply]
这个问题，已经解决了。是gitmodules引用主题的问题，根据作者的分享，已经搭建完成自己的Blog [hui61.com](https://hui61.com) 感谢🙏

**Reply by ZimaBlueee** [reply]
请问你是怎么解决的呀？

**Reply by hui61** [reply]
只要正确引用了gitmodules，应该就不会报这一类错误了，你可以搜一搜gitmodules如何引用

**Comment by markyun** [comment]
有时候tags 显示不出来，是为啥？

**Comment by reflector-li** [comment]
请问作者使用的 hugo 是什么版本，现在0.125会有很多问题。

**Reply by guangzhengli** [reply]
我是 0.99.0，确实很久没更新 hugo 的版本了

**Reply by reflector-li** [reply]
非常喜欢这个主题，有计划更新一下吗？

**Comment by ghost** [comment]
你好，我想问一下，footer那块不是调用的absURL，为什么我的站点部署后链接显示http://localhost:1313/不是我的网站网址呢，这个问题该怎么办？麻烦了谢谢

**Reply by ghost** [reply]
我没问题了，应该是我之前修改哪里错了

**Comment by think-crow** [comment]
开源一个项目就要无底洞的负责维护，对作者真的是一个坑！
前两天的疑问给作者添麻烦了！
以下问题供同路道友一个参考：
1、github pages部署时Actions里面报错(需要在github设置里面赋创建分支的权利)
```
Push the commit or tag
/usr/bin/git push origin gh-pages
remote: Permission to username/username.github.io.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/username/username.github.io.git/': The requested URL returned error: 403
Error: Action failed with "The process '/usr/bin/git' failed with exit code 128"

```
2、注意把作者例子文章里面的twtter链接给删掉！不然也会报错
3、关于创建完运行后blog菜单二次跳转后数据就会丢失（困扰好久的问题）
content里面只留blog、art文件夹，多余的都删掉就正常了

好的主题不能就这样埋没了！

**Comment by YechielZheng** [comment]
本地遇到了“page Not Found”提示，请问是什么原因呢？

**Reply by think-crow** [reply]
不知道你的hugo.toml配置了没，及content下面需要对应的md文件的！

**Reply by think-crow** [reply]
可以了吗，朋友？不要放弃啊！

**Reply by YechielZheng** [reply]
> 可以了吗，朋友？不要放弃啊！

谢谢，最近才开始继续搞了，我重新搞一下，再看看

**Comment by YechielZheng** [comment]
我也遇到了404

JQMIGRATE: Migrate is installed with logging active, version 3.5.0
jquery-migrate-3.5.0.js:136 JQMIGRATE: jQuery is not compatible with Quirks Mode
migrateWarn @ jquery-migrate-3.5.0.js:136Understand this warningAI
jquery-migrate-3.5.0.js:138 console.trace
migrateWarn @ jquery-migrate-3.5.0.js:138
(anonymous) @ jquery-migrate-3.5.0.js:199
(anonymous) @ jquery-migrate-3.5.0.js:22
(anonymous) @ jquery-migrate-3.5.0.js:24
localhost/:1 
        
Failed to load resource: the server responded with a status of 404 (Not Found)

**Comment by Jiang-HongXin** [comment]
难受啊..

Configuration file: none
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
     Build Warning: Layout 'archives' requested in themes/hugo-theme-ladder/exampleSite/content/archives.md does not exist.
     Build Warning: Layout 'archives' requested in themes/hugo-theme-ladder/exampleSite/content/archives.zh.md does not exist.
     Build Warning: Layout 'art' requested in themes/hugo-theme-ladder/exampleSite/content/art.md does not exist.
     Build Warning: Layout 'art' requested in themes/hugo-theme-ladder/exampleSite/content/art.zh.md does not exist.
     Build Warning: Layout 'blog' requested in themes/hugo-theme-ladder/exampleSite/content/blog.md does not exist.
     Build Warning: Layout 'blog' requested in themes/hugo-theme-ladder/exampleSite/content/blog.zh.md does not exist.
  Liquid Exception: Invalid Date: '{}' is not a valid datetime. in /_layouts/default.html
             ERROR: YOUR SITE COULD NOT BE BUILT:
                    ------------------------------------
                    Invalid Date: '{}' is not a valid datetime.

**Comment by alohanathan** [comment]
请问，hugo的版本哪个能最好兼容这个主题呢？最新版本的有报错了

**Comment by zhou-fuyi** [comment]
你好，可以说一下这部分的内容吗？

- > 本博客已于 2025 年 3 月 30 日使用 Nextjs 重构，并且使用 OpenNext 部署在 Cloudflare Pages 上。

**Reply by guangzhengli** [reply]
因为 hugo 官方基本已经不怎么维护了，加上模板语言快成时代眼泪了，所以之前开源的 hugo-ladder-theme 也不再维护，我用 Nextjs 重构了整个博客网站，后续会开源

**Reply by zhou-fuyi** [reply]
很期待，我想要将我的博客升级为你的这样。之前拜读了你关于个人开发的技术栈接触到了Next.js，配合 TailwindCSS + Shadcn/ui，感觉很适合我这样不太懂前端页面实现的情况。

**Comment by GuoYuefei** [comment]
大家记得初始化git的子模块，然后使用对应版本的hugo。
感觉可以跟着操作一遍之后自己使用最新的theme和hugo搭一套
